internal/ethapi/api.go

```diff
@@ -724,11 +728,42 @@ func (s *PublicBlockChainAPI) GetBlockByNumber(ctx context.Context, number rpc.B
 func (s *PublicBlockChainAPI) GetBlockByHash(ctx context.Context, hash common.Hash, fullTx bool) (map[string]interface{}, error) {
 	block, err := s.b.BlockByHash(ctx, hash)
 	if block != nil {
-		return s.rpcMarshalBlock(ctx, block, true, fullTx)
+		result, err := s.rpcMarshalBlock(ctx, block, true, fullTx)
+		if err != nil {
+			return nil, err
+		}
+		addEthCompatibilityFields(ctx, result, s.b, block.Header())
+		return result, nil
 	}
 	return nil, err
 }
 
+// addEthCompatibilityFields seeks to work around the incompatibility of celo
+// and ethers.js (and potentially other web3 clients) by adding fields to our
+// rpc response that ethers.js depends upon.
+// See https://github.com/celo-org/celo-blockchain/issues/1945
+func addEthCompatibilityFields(ctx context.Context, block map[string]interface{}, b Backend, header *types.Header) {
+	hash := header.Hash()
+	numhash := rpc.BlockNumberOrHash{
+		BlockHash: &hash,
+	}
+	gasLimit := b.GetBlockGasLimit(ctx, numhash)
+	block["gasLimit"] = hexutil.Uint64(gasLimit)
+
+	// Providing nil as the currency address gets the gas price minimum for the native celo asset.
+	baseFee, err := b.GasPriceMinimumForHeader(ctx, nil, header)
+
+	// GasPriceMinimumForHeader can return an error + non nil baseFee, an error
+	// and nil base fee or a base fee and a nil error. In all cases where a
+	// base fee is returned we want to add it as a field and if there was an
+	// error and nil base fee we log the failure.
+	if err != nil && baseFee == nil {
+		log.Debug("Not adding baseFeePerGas to RPC response, failed to retrieve gas price minimum", "block", header.Number.Uint64(), "err", err)
+		return
+	}
+	block["baseFeePerGas"] = (*hexutil.Big)(baseFee)
+}
```

Comment:
Probably is not breaking ethers, but why are we not adding the `difficulty` too? It's a low hang fruit if we use the block number

Reply:
I'd rather keep this change to the minimum required for Ethers.js, because its kinda hacky (we now have 2 different block representations ) and then spend some more time figuring out how to handle the other fields properly.